internal/controller/dnsrecord_controller.go

```diff
@@ -294,13 +296,20 @@ func (r *DNSRecordReconciler) updateStatus(ctx context.Context, previous, curren
 			current.Status.WriteCounter++
 			metrics.WriteCounter.WithLabelValues(current.Name, current.Namespace).Inc()
 			logger.V(1).Info("Changes needed on the same generation of record")
+
+			// if it is the same gen, but we had ready cond set before - we don't want to keep the "long"
+			// validation time from before
+			if readyCond.Status == metav1.ConditionTrue {
+				requeueTime = randomizedValidationRequeue
+			} else {
+				requeueTime = exponentialRequeueTime(current.Status.ValidFor)
+			}
+		} else {
+			requeueTime = randomizedValidationRequeue
```

Comment:
minor: could this be made a little more readable with a the following change?

```go
requeueTime = exponentialRequeueTime(current.Status.ValidFor)
if !generationChanged(current) {
	current.Status.WriteCounter++
	metrics.WriteCounter.WithLabelValues(current.Name, current.Namespace).Inc()
	logger.V(1).Info("Changes needed on the same generation of record")

	// if it is the same gen, but we had ready cond set before - we don't want to keep the "long"
	// validation time from before
	if readyCond.Status == metav1.ConditionTrue {
		requeueTime = randomizedValidationRequeue
	}
}

```

Reply:
Changed. Yep, it is confusing enough. I tried separating it into its own logic (the `calculateRequeue()`), but it did not help. Comments should help, but even with them, it is not ideal.